### PR TITLE
Pay 4029-OIDC-Deployment

### PR DESCRIPTION
**Pull request by Jennings-HMCTS**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: config/owasp/suppressions.xml [hunk 1]
```diff
@@ -7,5 +7,26 @@
     </notes>
     <cve>CVE-2018-1258</cve>
   </suppress>
- <!-- Supression Group ends here -->
+
+  <suppress until="2021-07-07">
```
Comment (aravindan89) [inline config/owasp/suppressions.xml:11]: Looks like an outdated file. Do we need to suppress this?

File: config/pmd/ruleset.xml [hunk 2]
```diff
@@ -62,7 +67,12 @@
   </rule>
   <rule ref="category/java/design.xml/TooManyMethods">
     <properties>
-      <property name="maxmethods" value="20" />
+      <property name="maxmethods" value="25" />
```
Comment (aravindan89) [inline config/pmd/ruleset.xml:70]: Instead of changing the rules is it possible to refactor our code to fulfil this rule?

**Review by aravindan89 — CHANGES_REQUESTED** [review]

File: config/owasp/suppressions.xml [hunk 3]
```diff
@@ -7,5 +7,26 @@
     </notes>
     <cve>CVE-2018-1258</cve>
   </suppress>
- <!-- Supression Group ends here -->
+
```
Comment (YasharRahvar) [inline config/owasp/suppressions.xml:10]: Why we are adding so many CVEs. We only have <cve>CVE-2018-1258</cve> at the moment in master

File: config/owasp/suppressions.xml (hunk 1 above)
Comment (Jennings-HMCTS) [reply config/owasp/suppressions.xml:11]: Yes Aravind... these are false positives and dont have fix... they are added as part of OIDC

File: config/owasp/suppressions.xml (hunk 3 above)
Comment (Jennings-HMCTS) [reply config/owasp/suppressions.xml:10]: These CVEs appear as part of OIDC and they are false positives. These CVEs appear in their latest version as well

File: Jenkinsfile_CNP [hunk 4]
```diff
@@ -15,11 +15,12 @@ def app = "bulkscanning-api"
...
 def secrets = [
   'ccpay-${env}': [
-    secret('cmc-service-secret', 'S2S_SERVICE_SECRET'),
-    secret('citizen-oauth-client-secret', 'OAUTH2_CLIENT_SECRET'),
+    secret('cmc-service-secret', 'S2S_SERVICE_SECRET_CMC'),
+    secret('citizen-oauth-client-secret', 'OAUTH2_CLIENT_SECRET_CMC_CITIZEN'),
     secret('freg-idam-generated-user-email-pattern', 'GENERATED_USER_EMAIL_PATTERN'),
     secret('freg-idam-test-user-password', 'TEST_USER_PASSWORD'),
-    secret('gov-pay-keys-cmc', 'GOV_PAY_AUTH_KEY_CMC')
+    secret('gov-pay-keys-cmc', 'GOV_PAY_AUTH_KEY_CMC'),
```
[the remark this answers is not in the document]
Comment (Jennings-HMCTS) [reply Jenkinsfile_CNP]: removed it Yashar

File: Jenkinsfile_CNP [hunk 5]
```diff
@@ -15,11 +15,12 @@ def app = "bulkscanning-api"
 
 def secrets = [
   'ccpay-${env}': [
-    secret('cmc-service-secret', 'S2S_SERVICE_SECRET'),
-    secret('citizen-oauth-client-secret', 'OAUTH2_CLIENT_SECRET'),
+    secret('cmc-service-secret', 'S2S_SERVICE_SECRET_CMC'),
```
[the remark this answers is not in the document]
Comment (Jennings-HMCTS) [reply Jenkinsfile_CNP]: Removed it Yashar

File: config/pmd/ruleset.xml (hunk 2 above)
Comment (Jennings-HMCTS) [reply config/pmd/ruleset.xml]: refactored Aravind

**Review by YasharRahvar — APPROVED** [review]

**Review by aravindan89 — APPROVED** [review]

**Review by amitchidrewar1301 — APPROVED** [review]

**Pull request merged**